pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -875,6 +937,44 @@ func (gce *GCECloud) firewallNeedsUpdate(name, serviceName, region, ipAddress st
 	return true, false, nil
 }
 
+func (gce *GCECloud) ensureHttpHealthCheckFirewall(region string, hosts []*gceInstance, hcName string, hcPort int32, isNodesHealthCheck bool) error {
+	// Prepare the firewall params for creating / checking.
+	desc := "Firewall for health checks from load-balancer."
+	sourceRanges := lbSrcRngsFlag.ipn
+	ports := []v1.ServicePort{{Protocol: "tcp", Port: hcPort}}
+
+	clusterId, err := gce.ClusterId.GetId()
+	if err != nil {
+		return fmt.Errorf("error getting cluster ID: %v", err)
+	}
+	fwNameSuffix := MakeHealthCheckFirewallNameSuffix(clusterId, hcName, isNodesHealthCheck)
+	fwName := makeFirewallName(fwNameSuffix)
```

Comment:
Regarding the name of firewall for nodeport healthcheck (`k8s-{clusterid}-{namespace}-{name}-{shortid}-hc`), not sure how to generate it since your commit is not in yet?